### PR TITLE
Dependencies update - Closes #684

### LINKED ISSUE #684: Review and upgrade dependencies
Due to old dependencies, there is a risk of failing builds.
Especially the Angular and Babel deps should be updated carefully.

One problem with Angular is described and solved here: #681
Any version higher than `1.7.0` breaks the Market Watcher.

### PR DESCRIPTION
**Pull request by MichalTuleja**
### What was the problem?
Old dependencies caused many potential security holes and produced broken builds.

### How did I fix it?
The most important dependencies were replaced according to Snyk and npm reports.
NodeJS version was updated to 8.x.

### How to test it?
Perform API and e2e tests.

### Review checklist
- The PR solves #684
- All new features are covered with e2e tests
- All new code follows best practices

### CODE REVIEW

**Comment by fchavant** [comment]
@MichalTuleja I am still seeing a few deprecation warnings from `npm`:
```npm WARN deprecated karma-jenkins-reporter@0.0.2: Currently not maintained
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated formatio@1.2.0: This package is unmaintained. Use @sinonjs/formatio instead
npm WARN deprecated wrench@1.3.9: wrench.js is deprecated! You should check out fs-extra (https://github.com/jprichardson/node-fs-extra) for any operations you were using wrench for. Thanks for all the usage over the years.
```

**Review by ghost — COMMENTED** [review]
`Dockerfile` LGTM but there are still some npm warnings

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,8 @@
-FROM node:6 AS builder
+FROM node:8 AS builder
 
 RUN useradd --create-home lisk && \
+    apt-get update && \
+    apt-get install nasm && \
```
Comment (fchavant) [inline Dockerfile]: Use `apt-get --assume-yes install nasm`

**Comment by fchavant** [comment]
@MichalTuleja great job, the build looks much better; there are a few more warning/notices that look important though:

```npm WARN deprecated nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
npm WARN notice [SECURITY] lodash has the following vulnerability: 1 low. Go here for more details: https://nodesecurity.io/advisories?search=lodash&version=3.7.0 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
npm WARN notice [SECURITY] tunnel-agent has the following vulnerability: 1 moderate. Go here for more details: https://nodesecurity.io/advisories?search=tunnel-agent&version=0.4.3 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
```
```
npm notice created a lockfile as package-lock.json. You should commit this file.
```
```
npm WARN leaflet.markercluster@1.3.0 requires a peer of leaflet@~1.0.3 but none is installed. You must install peer dependencies yourself.
```

**Review by ghost — APPROVED** [review]

**Pull request merged**